### PR TITLE
[tensorflow][build, test][ec2] Update smdebug binary to 0.9.2 for TF 2.3.0

### PR DESCRIPTION
**Pull request by NihalHarish**
*Issue #, if available:*

## Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [build] | [test] | [build, test] | [ec2, ecs, eks, sagemaker]
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [x] (If applicable) I've documented below the tests I've run on the DLC image
- [x] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [x] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

*Description:*
- Update smdebug to latest version (0.9.2) for TF2.3
- Skip gpu test on p2.8xlarge due to flakiness

*Tests run:*
- PR tests

*DLC image/dockerfile:*
tensorflow/training/docker/2.3.0/py3/cu102/Dockerfile.gpu
tensorflow/training/docker/2.3.0/py3/Dockerfile.cpu

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/ec2/test_smdebug.py [hunk 1]
```diff
@@ -18,6 +18,8 @@
 @pytest.mark.parametrize("ec2_instance_type", SMDEBUG_EC2_GPU_INSTANCE_TYPE, indirect=True)
 def test_smdebug_gpu(training, ec2_connection, region, gpu_only, py3_only):
     # TODO: Remove this once test timeout has been debugged (failures especially on p2.8xlarge)
+    if is_tf2(training) and "2.3.0" in training and SMDEBUG_EC2_GPU_INSTANCE_TYPE == "p2.8xlarge":
```
Comment (YYStreet) [inline test/dlc_tests/ec2/test_smdebug.py:21]: This line might not be correct. Should be `if "p2.8xlarge" in SMDEBUG_EC2_GPU_INSTANCE_TYPE:`

Comment (YYStreet) [reply test/dlc_tests/ec2/test_smdebug.py:21]: `get_ec2_instance_type()` returns a list

Comment (TusharKanekiDey) [reply test/dlc_tests/ec2/test_smdebug.py:21]: +1

Comment (arjkesh) [reply test/dlc_tests/ec2/test_smdebug.py:21]: good catch zhuo

Comment (saimidu) [inline test/dlc_tests/ec2/test_smdebug.py:21]: The condition for `is_tf2(image_uri)` is just a more generic version of `"2.3.0" in image_uri`.

Can make do with just the "2.3.0" check.

Comment (saimidu) [reply test/dlc_tests/ec2/test_smdebug.py]: In more detail -- `is_tf2(image_uri)` checks if the image version matches the format "2.x.y".

https://github.com/aws/deep-learning-containers/blob/e548c7566eb5b5b9d8aeeff8a28f275f66178996/test/test_utils/__init__.py#L66-L69

Comment (arjkesh) [reply test/dlc_tests/ec2/test_smdebug.py]: it also adds the "tensorflow" condition though which is important. I don't think it hurts here.

Comment (saimidu) [reply test/dlc_tests/ec2/test_smdebug.py]: > it also adds the "tensorflow" condition though which is important. I don't think it hurts here.

Alright. I'll edit my comment above. I should have put in "nit:" ahead of my comment.

**Pull request merged**